com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java

```diff
@@ -264,14 +264,21 @@ protected void _toJavaStatement(final XIssueExpression expr, final ITreeAppendab
       b.newLine();
     }
 
-    // checking for null context EObject
-    b.append("if (");
+    boolean issueExpressionEqualsImplicitVariable = true;
     if (markerObject != null) {
-      internalToConvertedExpression(markerObject, b, getLightweightType(eObjectType));
-    } else {
-      b.append(getContextImplicitVariableName(expr));
+      issueExpressionEqualsImplicitVariable = markerObject.toString().equalsIgnoreCase(getContextImplicitVariableName(expr));

```

Comment:
I don't think equalsIgnoreCase is what we want here (but can fix this in a followup)

Reply:
Done.